proof_ics23.go

```diff
@@ -166,3 +178,38 @@ func convertVarIntToBytes(orig int64, buf [binary.MaxVarintLen64]byte) []byte {
 	n := binary.PutVarint(buf[:], orig)
 	return buf[:n]
 }
+
+// GetWithProof gets the proof for the given key.
+func (t *ImmutableTree) GetWithProof(key []byte) (*ics23.CommitmentProof, error) {
+	if t.root == nil {
+		return nil, fmt.Errorf("cannot generate the proof with nil root")
+	}
+	value, err := t.Get(key)
+	if err != nil {
+		return nil, err
+	}
+	if value != nil {
+		return t.GetMembershipProof(key)
+	}
+	return t.GetNonMembershipProof(key)
```

Comment:
Is this correct? I think a non membership proof is about the key not existing, not the value. If the key doesn't exist, we will return an err on 189 instead of a non membership proof

Reply:
OK, I added `isExist` param to proof of both types (existence and non-existence).

Reply:
Seems like the correct fix is:

```go
value, err := t.Get(key)
if value != nil {
    return t.GetMembershipProof(key)
}
return t.GetNonMembershipProof(key)
```